src/mod_roster.erl

```diff
@@ -75,6 +75,30 @@
 
 -export([config_metrics/1]).
 
+-export([initialise_state/2,
```

Comment:
Maybe it makes sense to put the logic into a separate module?
To keep mod_roster reasonably sized.